### ISSUE TITLE
Regression report should have option to generate regression against particular checkstyle release

### ISSUE THREAD

**Issue by Abhishek-kumar09** [checkstyle/checkstyle#9367]
I have downloaded the latest cli from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words
_CLI not needed_

**How it works Now:**
The diff regression is always generated against the latest code of master branch.

**Is your feature request related to a problem? Please describe.**
Whenever we create a new regression report it always get created against  the latest master branch which is fine in most of the cases, but sometimes we require to generate regression against previous releases.

e.g. If we break something with a PR that is merged, if we want to correct those changes in new PR, then there is no sense in generating diff report against the latest master branch, instead what we could have done is like generated a diff against some previous commits or previous releases of checkstyle.

**Describe the solution you'd like**
We can modify the cli for generating diff report to take a commit or checkstyle release, 
Currently cli is only accepting these arguments: 
https://github.com/checkstyle/contribution/tree/master/checkstyle-tester#command-line-arguments

Also, If we could have option to generate diff report against the particular checkstyle releases.

**Comment by strkkk** [comment]
Each checkstyle release has a tag, and it looks like this line
https://github.com/checkstyle/checkstyle/blob/feceaccd77173d07d31e08b72eb3d8f6d1650e5b/.github/workflows/diff_report.yml#L176
can be changed to use tag instead of "master" (which will be value by default)
But it requires some testing :)

**Comment by nmancus1** [comment]
@Abhishek-kumar09 can you provide an example of where this is/ was necessary? Also, can you share how you would like to trigger such an action?  Perhaps `Base branch: {{commit numberl}}` ?

**Comment by Abhishek-kumar09** [comment]
> @Abhishek-kumar09 can you provide an example of where this is/ was necessary? Also, can you share how you would like to trigger such an action? Perhaps `Base branch: {{commit number}}` ?

This was the PR https://github.com/checkstyle/checkstyle/pull/9261 (first PR) that I made that was giving solving wrong indentation level for try handler and its `new` child, This PR got merged and checkstyle is released, After release user found some regression/ problem  with that release, so to solve that problem I created another PR https://github.com/checkstyle/checkstyle/pull/9362 (second PR)


Now the thing is the last PR created is deeply connected to first PR and it is solving the issue that was adressed in the first PR. There is no point of generating regression report with latest code, but what is needed is to generate a regression report with the commit before the first PR got merged, since it is in true sense solving the problem that first PR needed to solve.

**Comment by strkkk** [comment]
There is a reasonable workaround - generate report on local environment, it is a fallback for some non-standard cases.
I have been thinking about this and it seems that it is rare case and not useful for most of the users.

**Comment by nmancus1** [comment]
> I have been thinking about this and it seems that it is rare case and not useful for most of the users.

I agree, I think it is good to keep the documentation and usage of Github action simple.  If someone needs specific reports like this, they are likely already familiar with the project enough that running `diff.groovy` with a different base branch would be trivial.
